src/gtest/test_checkblock.cpp

```diff
@@ -110,292 +182,156 @@ TEST(ContextualCheckBlock, BadCoinbaseHeight) {
     EXPECT_TRUE(ContextualCheckBlock(block, state, &indexPrev));
 }
 
+// TEST PLAN: first, check that each ruleset accepts its own transaction type.
+// Currently (May 2018) this means we'll test Sprout-Sprout,
+// Overwinter-Overwinter, and Sapling-Sapling.
 
-// Test that a block evaluated under Sprout rules cannot contain Sapling transactions.
+// Test block evaluated under Sprout rules will accept Sprout transactions.
 // This test assumes that mainnet Overwinter activation is at least height 2.
-TEST(ContextualCheckBlock, BlockSproutRulesRejectSaplingTx) {
-    SelectParams(CBaseChainParams::MAIN);
+TEST_F(ContextualCheckBlockTest, BlockSproutRulesAcceptSproutTx) {
+    CMutableTransaction mtx = GetFirstBlockTransaction();
 
-    CMutableTransaction mtx;
-    mtx.vin.resize(1);
-    mtx.vin[0].prevout.SetNull();
-    mtx.vin[0].scriptSig = CScript() << 1 << OP_0;
-    mtx.vout.resize(1);
-    mtx.vout[0].scriptPubKey = CScript() << OP_TRUE;
-    mtx.vout[0].nValue = 0;
-
-    mtx.fOverwintered = true;
-    mtx.nVersion = SAPLING_TX_VERSION;
-    mtx.nVersionGroupId = SAPLING_VERSION_GROUP_ID;
-
-    CTransaction tx {mtx};
-    CBlock block;
-    block.vtx.push_back(tx);
-
-    MockCValidationState state;
-    CBlockIndex indexPrev {Params().GenesisBlock()};
+    // Make it a Sprout transaction w/o JoinSplits
```

Comment:
It's a coinbase transaction, so it can never (under current consensus rules) contain JoinSplits, SpendDescriptions or OutputDescriptions.